### PR TITLE
Fixes #15979 - Add API for Ansible roles

### PR DESCRIPTION
**Pull request by xprazak2**
Needs #25 to work.

### CODE REVIEW

File: app/controllers/api/v2/ansible_roles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Api
+  module V2
+    # API controller for Ansible Roles
+    class AnsibleRolesController < ::Api::V2::BaseController
+      include ::Api::Version2
+
+      before_action :find_resource, :only => [:destroy]
+
+      api :GET, '/ansible/ansible_roles', N_('List Ansible roles')
```
Comment (dLobatog) [inline app/controllers/api/v2/ansible_roles_controller.rb]: I'm missing an endpoint for importing here

Comment (dLobatog) [inline app/controllers/api/v2/ansible_roles_controller.rb]: `/ansible/ansible_roles/:id`

File: app/controllers/api/v2/ansible_roles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module Api
+  module V2
+    # API controller for Ansible Roles
+    class AnsibleRolesController < ::Api::V2::BaseController
+      include ::Api::Version2
+
+      before_action :find_resource, :only => [:destroy]
```
Comment (dLobatog) [inline app/controllers/api/v2/ansible_roles_controller.rb]: `index` and `import` missing

File: app/controllers/api/v2/ansible_roles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+      api :DELETE, '/ansible/ansible_role/:id', N_('Deletes Ansible role')
+      param :id, :identifier, :required => true
+      def destroy
+        process_response @ansible_role.destroy
+      end
+
+      private
+
+      def find_resource
```
Comment (dLobatog) [inline app/controllers/api/v2/ansible_roles_controller.rb]: Just wondering why can you not rely on the `find_resource` provided by core?

**Comment by dLobatog** [comment]
@xprazak2 Thanks, see comments inline. This should be easy to merge :smile:

**Comment by xprazak2** [comment]
I use `find_resource` in controller and I added `show`, `import` and `obsolete` actions. Because we cannot do a two step import as in UI, `import` imports all missing roles from proxy (or Foreman instance if no proxy is given in params), same goes for `obsolete`.

File: app/controllers/api/v2/ansible_roles_controller.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        process_response @ansible_role.destroy
+      end
+
+      api :POST, '/ansible/ansible_roles/import', N_('Import Ansible roles')
+      param :proxy, Hash, N_('Smart Proxy to import from')
+      def import
+        @imported = @importer.import!
+      end
+
+      api :POST, '/asnible/ansible_roles/obsolete', N_('Obsolete Ansible roles')
```
Comment (dLobatog) [inline app/controllers/api/v2/ansible_roles_controller.rb]: typo, I'll fix it on merge.

**Comment by dLobatog** [comment]
Merged as 3a7067dcd61b5775ed0309c1006ea82301ef3373, thanks @xprazak2

**Pull request closed without merging**